core/shared/src/main/scala/zio/stm/TArray.scala

```diff
@@ -17,138 +17,193 @@
 package zio.stm
 
 /** Wraps array of [[TRef]] and adds methods for convenience.
- * Caution: most of methods are not stack-safe.
- * */
-class TArray[A] private (val array: Array[TRef[A]]) extends AnyVal {
+ *
+ */
+class TArray[A] private (private val array: Array[TRef[A]]) extends AnyVal {
```

Comment:
Should we make this and the other `STM` types `final`? Technically classes extending `AnyVal` are already final but could be good to clarify intent. Looks like we do this elsewhere in the code base.

Reply:
Absolutely agreed, done. Regarding the other "leakings", I'll tackle them in a separate PR(s), just to preserve the scope of this one.

Reply:
@mijicd This could also potentially be good first PRs. :D

Reply:
@saraiva132 I'll double-check if it makes sense to do such change, and if so, I'll make some issues!